Source/TheAviator/gherkin/ScenarioEnvironment.ts

```diff
@@ -12,47 +12,58 @@ import { Scenario } from './Scenario';
 import { ISerializer } from '../ISerializer';
 
-export type MicroserviceMethod = (microservice: Microservice) => Promise<void>;
+const eventStoreDumpFolderName = 'eventStore';
+const containerOptionsFileExtension = '.json';
+
+export type MicroserviceAction = (microservice: Microservice) => Promise<void>;
 
 export class ScenarioEnvironment {
-    static empty: ScenarioEnvironment = new ScenarioEnvironment({} as IFlightPaths, new ScenarioEnvironmentDefinition(), {}, {} as ISerializer);
 
-    readonly definition: ScenarioEnvironmentDefinition;
-    readonly microservices: { [key: string]: Microservice };
+    /**
+     * Gets an empty {ScenarioEnvironment}
+     *
+     * @static
+     * @type {ScenarioEnvironment}
+     */
+    static readonly empty: ScenarioEnvironment = new ScenarioEnvironment(
+        {} as IFlightPaths,
+        new ScenarioEnvironmentDefinition(),
+        {},
+        {} as ISerializer);
 
     constructor(
-        private _flightPaths: IFlightPaths,
-        definition: ScenarioEnvironmentDefinition,
-        microservices: { [key: string]: Microservice },
+        private readonly _flightPaths: IFlightPaths,
+        readonly definition: ScenarioEnvironmentDefinition,
+        readonly microservices: { [key: string]: Microservice },
         readonly _serializer: ISerializer) {
-        this.definition = definition;
-        this.microservices = microservices;
         this.writeConfigurationFiles();
     }
 
     async start(): Promise<void> {
-        await this.forEachMicroservice(async _ => await _.start());
+        await this.forEachMicroservice(_ => _.start());
         await this.connectConsumersToProducers();
     }
 
     async stop(): Promise<void> {
         await this.disconnectConsumersFromProducers();
-        await this.forEachMicroservice(async _ => await _.kill());
+        await this.forEachMicroservice(_ => _.kill());
     }
 
-    async forEachMicroservice(method: MicroserviceMethod) {
+    forEachMicroservice(action: MicroserviceAction): Promise<void[]> {
+        const promises: Promise<void>[] = [];
         for (const microservice of Object.values(this.microservices)) {
-            await method(microservice);
+            promises.push(action(microservice));
         }
+        return Promise.all(promises);
     }
 
     async dumpEventStore(scenario: Scenario) {
         this.forEachMicroservice(async (microservice) => {
             const microserviceDestinationDirectory = this._flightPaths.forMicroserviceInScenario(scenario, microservice);
-            const destinationDirectory = path.join(microserviceDestinationDirectory, 'eventStore');
-
-            if (!fs.existsSync(destinationDirectory)) {
-                fs.mkdirSync(destinationDirectory, { recursive: true });
+            const destinationDirectory = path.join(microserviceDestinationDirectory, eventStoreDumpFolderName);
+            const directoryExists = await promisify(fs.exists)(destinationDirectory);
```

Comment:
FS has a promises object on it with async promise based versions. So you can skip the promisify all together.
(https://nodejs.org/api/fs.html#fs_fs_promises_api)

```suggestion
            const directoryExists = await fs.promises.exists(destinationDirectory);
```